Generate an Access Monitoring event reference

ptgott opened the pull request:
Closes #60074

Use the getters we declared to access protobuf-generated event structs in order to create Athena views for Access Monitoring events, populating a template with the event structs and including examples of `tctl audit query exec` queries.

Add Make targets to generate the page and check that it was generated. Add a step to the `Lint (Proto)` job to check that this was generated.

ptgott committed:
Generate an Access Monitoring event reference

Closes #60074

Use the getters we declared to access protobuf-generated event structs
in order to create Athena views for Access Monitoring events, populating
a template with the event structs and including examples of `tctl audit
query exec` queries.

Add Make targets to generate the page and check that it was generated.
Add a step to the `Lint (Proto)` job to check that this was generated.

zmb3 commented on gen-athena-docs/main.go:
```diff
@@ -0,0 +1,118 @@
+	for i := 0; i < len(cols) && i < 3; i++ {
+		names = append(names, cols[i].NameSQL())
+	}
+	return strings.Join(names, ",")
```
It doesn't matter at all in this case since `i` is so small, but for illustration's sake here's another way you could write this that doesn't require allocating a slice that will be thrown away.

```suggestion
	var sb strings.Builder
	for i := range min(3, len(cols)) {
	    if i != 0 {
	        sb.WriteString(",")
	    }
	    sb.WriteString(cols[i].NameSQL())
	}
	return sb.String()
```

zmb3 commented on build.assets/tooling/go.mod:
```diff
@@ -78,9 +78,273 @@ require (
 	sigs.k8s.io/yaml v1.6.0 // indirect
 )
 
+require github.com/gravitational/teleport v0.0.0-00010101000000-000000000000
```
Best to move this into the main `require` block up above.

zmb3 commented on build.assets/tooling/go.sum:
The dependency on `teleport` adds a lot of bloat here, but it's probably fine.

zmb3 reviewed (approved):
Nice work

camscale commented on build.assets/tooling/cmd/gen-athena-docs/main.go:
```diff
@@ -0,0 +1,109 @@
+	// Capitalize the first word in the description.
+	if len(desc) > 1 {
+		desc = strings.ToUpper(string(desc[0])) + desc[1:]
+	}
```
This is not unicode/utf-8 safe as it can split the rune if it is not 8-bits. You need to do something like:
```
if r, size := utf8.DecodeRuneInString(text); r != utf8.RuneError {
    s = string(unicode.ToUpper(r)) + text[size:]
}
```

ptgott committed:
Clean up the event schema reference generator

- Remove unnecessary line breaks.
- Move `teleport` requirement into the main `require` block.
- Embed the template.
- Accommodate characters that exceed one byte in `prepareDescription`.
- Use a no-allocation alternative to strings.Join in `colNameList`.
